src/models/invites-ignorer.ts

```diff
@@ -215,6 +230,19 @@ export class IgnoredInvites {
      * other concurrent rewrites of the same object.
      */
     public async getOrCreateTargetRoom(): Promise<Room> {
+        // Synchronous code. Do NOT introduce any `await` before locking
+        // or there will be race conditions on both in-memory data and
+        // homeserver-stored data.
+        //
+        // Thanks to run-to-completion, the uninterruptible behavior of this
+        // method is the following:
```

Comment:
This comment (and the lines which follow) aren't really describing anything: it's just saying what we're about to do, not why it's important to have a lock. We know what it's about to do: there's code to tell us :)

Reply:
We should also describe why we don't need cross-client locking, considering this code is seemingly important.

It's further a bad smell that we need a lock at all - why would this code be called multiple times?

Reply:
> This comment (and the lines which follow) aren't really describing anything: it's just saying what we're about to do, not why it's important to have a lock. We know what it's about to do: there's code to tell us :)

...because of race conditions? I'm reading, modifying, writing some user account data, that's inherently racy.

> We should also describe why we don't need cross-client locking, considering this code is seemingly important.

Wait, we have cross-client locking? Where do I find the documentation on these?

> It's further a bad smell that we need a lock at all - why would this code be called multiple times?

I come from a safety background. Any code is considered to be called multiple times unless proven otherwise :)